artemis-server/src/main/java/org/apache/activemq/artemis/spi/core/security/jaas/kubernetes/client/KubernetesClientImpl.java

```diff
@@ -157,18 +161,12 @@ private SSLContext buildSSLContext() throws Exception {
          logger.debug("Kubernetes CA certificate not found at: {}. Truststore not configured", caPath);
          return ctx;
       }
-      try (InputStream fis = new FileInputStream(certFile)) {
-         KeyStore trustStore = KeyStore.getInstance(KeyStore.getDefaultType());
-         CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
-         X509Certificate certificate = (X509Certificate) certFactory.generateCertificate(fis);
-         trustStore.load(null, null);
-         trustStore.setCertificateEntry(certFile.getName(), certificate);
-         TrustManagerFactory tmFactory = TrustManagerFactory
-               .getInstance(TrustManagerFactory.getDefaultAlgorithm());
-         tmFactory.init(trustStore);
-
-         ctx.init(null, tmFactory.getTrustManagers(), new SecureRandom());
-      }
+      KeyStore trustStore = SSLSupport.loadKeystore(null, "PEMCA", caPath, null);
```

Comment:
this looks good. I think it is the correct approach.

Reply:
Thanks!